### PR TITLE
update plaid-node to match 2019-05-29 API release

### PR DESCRIPTION
**Pull request by tanatplaid**
* identity to have owners on each account
* transaction/identity address changes
* add international/bacs numbers
* bump version
* remove contact info from institutions response, as it seems unused.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -327,6 +325,18 @@ declare module 'plaid' {
...
   }
 
+  interface InternationalNumbers {
+    account_id: string;
+    iban: string;
+    bic: string;
+  }
+
+  interface BACSNumbers {
+    bacs: string;
```
Comment (jasonjia93) [inline index.d.ts]: This should be `account_id`

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "plaid",
-  "version": "3.1.1",
+  "version": "4.0.0",
```
Comment (ntindall) [inline package.json]: we don't need to manually bump this.

File: index.d.ts [hunk 3]
```diff
@@ -457,7 +468,7 @@ declare module 'plaid' {
   }
 
   interface ClientOptions extends CoreOptions {
-    version?: '2018-05-22' | '2017-03-08';
+    version?: '2019-05-29';
```
Comment (ntindall) [inline index.d.ts]: I think we need to change the default version somewhere else in the client too? this is only the typescript typings, not the actual implementation.

File: index.d.ts (hunk 1 above)
Comment (tanatplaid) [reply index.d.ts]: ⚡️

File: package.json (hunk 2 above)
Comment (tanatplaid) [reply package.json]: ⚡️

File: index.d.ts (hunk 3 above)
Comment (tanatplaid) [reply index.d.ts]: i can't see it set anywhere; it looks like you have to explicitly set it in `options` or it uses the default on theserver.

Comment (jasonjia93) [reply index.d.ts]: Looking at this: https://plaid.com/docs/api-upgrades/#versioning
It doesn't sound like the node client is supposed to have a default, it has to be initialized with a specific version

File: test/PlaidClientTest.js [hunk 4]
```diff
@@ -643,11 +648,7 @@ describe('plaid.Client', () => {
           expect(response).to.be.ok();
           expect(response.report).to.be.ok();
 
-          // The transactions in an Asset Report with Insights should have a
```
Comment (ntindall) [inline test/PlaidClientTest.js]: why is this change necessary?

Comment (tanatplaid) [reply test/PlaidClientTest.js]: i think the accounts returned are non-deterministic 
it's hard for my to debug this further without having the credentials CI has.

File: index.d.ts (hunk 3 above)
Comment (ntindall) [reply index.d.ts]: Interesting. does that makes sense?

I know the api will default to using the client version they have configured on the dashboard. But consider the case where a client has an older version configured, and then upgrades their node version. They'd expect to get API responses conforming with the new typescript typings, but this wouldn't be the case until they supplied the version explicitly, which could be difficult to debug.

I think we should add `2019-05-29` it as the default option if its not supplied. I think this is the behavior of the other clients... Thoughts @tanatplaid @jasonjia93 ?

Comment (tanatplaid) [reply index.d.ts]: :zap: we've added a default option which should make sense with a version upgrade

File: index.d.ts [hunk 5]
```diff
@@ -139,13 +144,7 @@ declare module 'plaid' {
   }
 
   interface InstitutionWithContactData extends Institution {
-    addresses: Array<{
```
Comment (ntindall) [inline index.d.ts]: @tanatplaid we can actually remove `InstitutionWithContactData` - it is no longer possible to get this from the API.

File: index.d.ts [hunk 6]
```diff
@@ -172,13 +171,6 @@ declare module 'plaid' {
     phone_numbers: Array<PhoneNumber>;
   }
 
-  interface AccountIdentity {
```
Comment (ntindall) [inline index.d.ts]: We still need this schema for the `Asset` related endpoints - I think we can add typings to the effect of `AssetIdentity` 

the address schema has not changed for the assets product, only for everything else (`/accounts` `/identity`).

File: index.d.ts (hunk 3 above)
Comment (ntindall) [reply index.d.ts]: 👍

File: index.d.ts [hunk 7]
```diff
@@ -172,29 +177,18 @@ declare module 'plaid' {
...
-    phone_numbers: Array<PhoneNumber>;
-  }
-
   interface Address {
     accounts: Array<string>;
     data: AddressData;
     primary: boolean;
   }
 
-  interface AccountAddress {
```
Comment (tanatplaid) [inline index.d.ts]: this was unused.

File: index.d.ts (hunk 6 above)
Comment (tanatplaid) [reply index.d.ts]: :zap:

File: index.d.ts (hunk 5 above)
Comment (tanatplaid) [reply index.d.ts]: ⚡️

File: index.d.ts [hunk 8]
```diff
@@ -66,6 +66,10 @@ declare module 'plaid' {
     };
   }
 
+  interface AccountWithIdentity extends Account {
```
Comment (ntindall) [inline index.d.ts]: nit: can we call this `AccountWithOwners` ?

File: index.d.ts [hunk 9]
```diff
@@ -354,11 +389,10 @@ declare module 'plaid' {
   }
 
   interface IncomeResponse extends AccountsResponse {
-    income: Income;
```
Comment (ntindall) [inline index.d.ts]: why was this removed @tanatplaid ?

File: index.d.ts (hunk 8 above)
Comment (jasonjia93) [reply index.d.ts]: +1 to be consistent with the API

File: index.d.ts (hunk 9 above)
Comment (tanatplaid) [reply index.d.ts]: mistake

File: index.d.ts [hunk 10]
```diff
@@ -284,7 +268,32 @@ declare module 'plaid' {
     days_available: number;
     historical_balances: Array<HistoricalBalance>;
     transactions: Array<AssetReportTransaction>;
-    owners: Array<Identity>;
+    owners: Array<AssetIdentity>;
+  }
+
+  // Different from "Identity", as it belongs with "Assets"
+  // which is only for US.
+  interface AssetIdentity {
```
Comment (ntindall) [inline index.d.ts]: this is a nit: but can we call this `AssetsIdentity` (and `AssetsAddress` / `AssetsAddressData` ) 

I think the name ideally aligns with the product name.
